### PR TITLE
Implement the Bazel building recipe as a stand-alone script that can be invoked from build.sh.

### PR DESCRIPTION
**Pull request by stefanbucur**
Before I update the documentation and other Bazel projects, I wanted to get your early input on the approach.

### CODE REVIEW

**Comment by inferno-chromium** [comment]
> Before I update the documentation and other Bazel projects, I wanted to get your early input on the approach.

SGTM, can start doing that in followup PRs, this approach sgtm.

**Comment by catenacyber** [comment]
Looks good.
Would like to see the change in existing projects in the same PR as well ;-)

File: infra/base-images/base-builder/bazel_build_fuzz_tests [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+declare -r OSS_FUZZ_TESTS=(
+    $(bazel query "${BAZEL_FUZZ_TEST_QUERY}" | sed "s/$/${BAZEL_PACKAGE_SUFFIX}/")
+)
+
+echo "Found ${#OSS_FUZZ_TESTS[@]} fuzz test packages:"
+for oss_fuzz_test in "${OSS_FUZZ_TESTS[@]}"; do
+    echo "  ${oss_fuzz_test}"
+done
+
+declare -r BAZEL_BUILD_FLAGS=(
```
Comment (asraa) [inline infra/base-images/base-builder/bazel_build_fuzz_tests:43]: Will the script need to add in the clang ubsan libraries for undefined builds?

Comment (stefanbucur) [reply infra/base-images/base-builder/bazel_build_fuzz_tests:43]: Not at this point. It's being taken care of at the Bazel extension level: https://github.com/bazelbuild/rules_fuzzing/blob/fc0e7d437fda6666b9f5f25d1a40ae919cfcecf2/fuzzing/private/oss_fuzz/repository.bzl#L59-L69

But it's a good question whether this should become part of this standard script vs. it being integrated in the repository rule logic. Maybe as the library adoption grows, we'll better learn about the pros/cons of each approach.

Comment (asraa) [reply infra/base-images/base-builder/bazel_build_fuzz_tests:43]: Awesome, thank you!

**Pull request merged**